homeassistant/setup.py

```diff
@@ -167,16 +170,28 @@ def log_error(msg: str, link: Optional[str] = None) -> None:
 
     try:
         if hasattr(component, "async_setup"):
-            result = await component.async_setup(  # type: ignore
+            task = component.async_setup(  # type: ignore
                 hass, processed_config
             )
         elif hasattr(component, "setup"):
-            result = await hass.async_add_executor_job(
-                component.setup, hass, processed_config  # type: ignore
+            # This should not be replaced with hass.async_add_executor_job because
+            # we don't want to track this task in case it blocks startup.
+            task = hass.loop.run_in_executor(
+                None, component.setup, hass, processed_config  # type: ignore
             )
         else:
-            log_error("No setup function defined.")
+            _LOGGER.error("No setup function defined for %s", domain)
```

Comment:
Why switch to `log_error` ?

Reply:
Forgot to switch this back in testing.  Good catch. Reverted